llvm/lib/Target/SystemZ/SystemZTargetTransformInfo.cpp

```diff
@@ -74,6 +74,40 @@ static void countNumMemAccesses(const Value *Ptr, unsigned &NumStores,
     }
 }
 
+static void getNumGEPIndexUses(const Value *V, unsigned &NumGEPIdxUses) {
+  for (const User *U : V->users()) {
+    if (const auto *LI = dyn_cast<LoadInst>(U)) {
+      assert(isa<AllocaInst>(V) && LI->getType()->isIntegerTy() &&
+             "Expected a load only from the alloca, with integer type.");
```

Comment:
The first direct use of the alloca must be a load, right (cannot have sext, zext, or getelementptr on a pointer type).  Maybe this should be pulled out and done by the caller?

Also, should this function just return the number of users as return value?  In fact, the only caller of this function doesn't even care about the *number* of users, just whether there is one ... so maybe just a bool return?